R/print.data.table.R

```diff
@@ -111,37 +111,27 @@ print.data.table = function(x, topn=getOption("datatable.print.topn"),
     #   function below adds those back when necessary
     toprint = toprint_subset(toprint, cols_to_print)
   }
+  print_default = function(x) {
+    if (col.names != "none") { cut_colnames = identity }
+    cut_colnames(print(x, right=TRUE, quote=quote, na.print=na.print))
+    # prints names of variables not shown in the print
+    if (trunc.cols && length(not_printed) > 0L) trunc_cols_message(not_printed, abbs, class, col.names)
```

Comment:
let's redefine `trunc.cols` above:

```
if (isTRUE(trunc.cols)) {
  ...
  trunc.cols <- length(not_printed) > 0L
}
```

Then we can simplify the condition here: `if (trunc.cols) trunc_cols_message(...)`.

Alternatively, we can seed `not_printed=NULL`, then handle that directly in the `trunc_cols_message()` helper:

```r
not_printed=NULL
if (trunc.cols) {
  not_printed=...
}

trunc_cols_message=function(not_printed, ...) {
  if (!length(not_printed)) return(invisible())
  ...
}
```

Take a look at both approaches, my instinct is the latter will look more readable in the end.

Reply:
I think if we're just doing this to simplify the condition, the former method is cleaner IMO